File: Tema 6 - React/src/Models/Fundraiser.ts
import type { Person } from './Person';

export class Fundraiser {
  id?:number;
  name: string;
  goalValue: number;
  currentlyRaisedAmount?: number;
  dueDate: Date;
  creationDate?: Date;
  status?: Status;
  owner: Person;
  donors?: Person[];

  constructor(
    name: string,
    goalValue: number,
    dueDate: Date,
    owner: Person,
    id?:number,
    creationDate?: Date,
    status?: Status,
    currentlyRaisedAmount?: number,
    donors?: Person[],
  ) {
    this.name = name;
    this.goalValue = goalValue;
    this.currentlyRaisedAmount = currentlyRaisedAmount;
    this.creationDate = creationDate;
    this.dueDate = dueDate;
    this.status = status;
    this.owner = owner;
    this.donors = donors;
    this.id=id;
  }
}

export const enum Status {
  'Closed' = 'Closed',
  'Active' = 'Active',
}

File: Tema 5 - Typescript/after/PetShelterBackend/PetShelter.Api/Resources/Fundraiser.cs
using PetShelter.Domain;

namespace PetShelter.Api.Resources
{
    public class Fundraiser
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public decimal GoalValue { get; set; }
        public decimal CurrentDonationAmount { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime DueDate { get; set; }
        public Status Status { get; set; }
        public Person Owner { get; set; }
        public ICollection<Person> Donors { get; set; }
    }
}
